Fixed the bug where the speciality IO channels got broken on reloading the bitfile

zszilagy opened the pull request:
- [X] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/niveristand-fpga-addon-custom-device/blob/master/CONTRIBUTING.md).

### What does this Pull Request accomplish?

Provides a fix where the Speciality IO Channels got broken upon reloading the bitfile.
Provides a fix where the Default Values of the Speciality IO was not updated.
Corrects the misleading comments in the Bitfile Reload Unit tests.
Adds Speciality IO unit tests.

### Why should this Pull Request be merged?

This PR contains the handler of the Speciality IO channels for the Reload bitfile functionality.

### What testing has been done?

Unit tests were written where all the implementation was validated.

DStavilaNI commented:
Here is my feedback regarding this PR:

1. 
![image](https://user-images.githubusercontent.com/35329597/111975520-b2a53680-8ace-11eb-9847-e60db03304eb.png) - This is bad practice. The error wire should be passed through to check for data conversion errors.
2. 
![image](https://user-images.githubusercontent.com/35329597/111975692-e2ecd500-8ace-11eb-9f14-cce7e7b8b3ab.png) - this VI is called "Set Scalar Channel Data.vi", however it operates on both scalar channels and groups. Additionally, it does not "Set", it actually gets or generates/packages data coming from FPGA Addon channel or section items. I think it should be "Get Item Data.vi" or something like that.
3. 
![image](https://user-images.githubusercontent.com/35329597/111976371-8d64f800-8acf-11eb-9410-5b6e416a27d1.png) - again bad coding. We should check for errors on data conversion, not leave it hanging. Otherwise it is quite hard to debug. 
- What is the point of that sorting function ? Did you check If actually changing the order of the registers in the array does not lead to errors or unexpected behavior ?

4. 
![image](https://user-images.githubusercontent.com/35329597/111977546-d9646c80-8ad0-11eb-8a04-488a9bd607a6.png) ... I think an error case structure would be required here to check for errors before the execution of this code per recommended LV development practices.

5. 
![image](https://user-images.githubusercontent.com/35329597/111977930-568fe180-8ad1-11eb-89db-8e152e3308c9.png) - I am not sure this is best practice. This long wire is unnecessary and it does not help to make the code more readable.

zszilagy commented:
> Here is my feedback regarding this PR:
> 
> 
> ![image](https://user-images.githubusercontent.com/35329597/111975520-b2a53680-8ace-11eb-9847-e60db03304eb.png) - This is bad practice. The error wire should be passed through to check for data conversion errors.
> 2.
> ![image](https://user-images.githubusercontent.com/35329597/111975692-e2ecd500-8ace-11eb-9f14-cce7e7b8b3ab.png) - this VI is called "Set Scalar Channel Data.vi", however it operates on both scalar channels and groups. Additionally, it does not "Set", it actually gets or generates/packages data coming from FPGA Addon channel or section items. I think it should be "Get Item Data.vi" or something like that.
> 3.
> ![image](https://user-images.githubusercontent.com/35329597/111976371-8d64f800-8acf-11eb-9410-5b6e416a27d1.png) - again bad coding. We should check for errors on data conversion, not leave it hanging. Otherwise it is quite hard to debug.
> 
> * What is the point of that sorting function ? Did you check If actually changing the order of the registers in the array does not lead to errors or unexpected behavior ?
> 
> 
> ![image](https://user-images.githubusercontent.com/35329597/111977546-d9646c80-8ad0-11eb-8a04-488a9bd607a6.png) ... I think an error case structure would be required here to check for errors before the execution of this code per recommended LV development practices.
> 
> 
> ![image](https://user-images.githubusercontent.com/35329597/111977930-568fe180-8ad1-11eb-89db-8e152e3308c9.png) - I am not sure this is best practice. This long wire is unnecessary and it does not help to make the code more readable.

1. Fixed as requested.
2. Fixed as requested.
3. Fixed as requested. 
    NOTE: the same method was applied as in the Create Write Channel.vi. As the 'default value' here represents a list of channels which will end up in the Available registers table in the System Explorer, it's necessary to sort the channels alphabetically. Not having the sort in the "Group" case will add the channels as they are and it will not resemble the original functionality.
4. Fixed as requested.
5. Fixed as requested.
